libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java

```diff
@@ -512,17 +535,26 @@ public synchronized void setPushServiceType(Class<? extends PushService> type) {
     }
 
     /**
-     * Returns the passcode manager that's associated with SalesforceSDKManager.
+     * Returns the descriptor of the passcode activity class that's currently in use.
+     *
+     * @return Passcode activity class descriptor.
+     *
+     * @deprecated Will be removed in Mobile SDK 10.0.  Use {@link #getScreenLockManager()} instead.
+     */
+    public PasscodeManager getPasscodeManager() { return new PasscodeManager(context); }
+
+    /**
+     * Returns the ScreenLock manager that's associated with SalesforceSDKManager.
      *
-     * @return PasscodeManager instance.
+     * @return ScreenLockManager instance.
      */
-    public PasscodeManager getPasscodeManager() {
-    	synchronized (passcodeManagerLock) {
-            if (passcodeManager == null) {
-                passcodeManager = new PasscodeManager(context);
+    public ScreenLockManager getScreenLockManager() {
```

Comment:
Do  we still need this? Remove it if possible.

Reply:
I believe we do since we still need to set mobile policy in `OAuthWebviewHelper` and update on unlock from `ScreenLockActivity`.  Unfortunately both of those are in the `ui` package so it needs to remain public.  Also we also need it for user cleanup in `SalesforceSDKManager` so I don't think it makes sense to move it.